### PR TITLE
feature/pure cURL http client

### LINKED ISSUE #16: Being able to add some integration version to user-agent
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE

**What happened**:
Today the user-agent contains: curl / openssl / weglot versions.

**What you expected to happen**:
We should give the user a way to add the integration version for debugging purposes.

### LINKED ISSUE #19: Client: Replace Guzzle with pure cURL
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE

**What happened**:
We're actually using Guzzle for communication purposes with our API. It use cURL internally.

**What you expected to happen**:
Using only cURL by ourself, like that we'll be able to custom cURL options easily

**Anything else we need to know?**:
Let's try to take a look at https://github.com/stripe/stripe-php/blob/master/lib/HttpClient/CurlClient.php

### PR DESCRIPTION
**Pull request by Korbeil**
**What this PR does / why we need it**:
Replacing Guzzle with pure cURL and interface for further HTTP clients.

**Which issue(s) this PR fixes**:

- Fixes #19 
- Fixes #16 

**Special notes for your reviewer**:
The main class of this PR: `Weglot\Client\HttpClient\CurlClient` is mostly based on Stripe cURL integration: https://github.com/stripe/stripe-php/blob/master/lib/HttpClient/CurlClient.php

**How you would summarize your PR ? (for changelog)**:
(changelog delivered with the feature)

### CODE REVIEW

File: src/Client/Endpoint/Status.php [hunk 1]
```diff
@@ -19,9 +19,9 @@ class Status extends Endpoint
      */
     public function handle()
     {
-        $response = $this->request([], false);
+        list($rawBody, $httpStatusCode, $httpHeader) = $this->request([], false);
```
Comment (Gmulti) [inline src/Client/Endpoint/Status.php]: It's ok with difference between php5 and php7 ?

Comment (Korbeil) [reply src/Client/Endpoint/Status.php]: `list()` was introduced in php4
(cf. http://php.net/manual/fr/function.list.php)

Comment (Gmulti) [reply src/Client/Endpoint/Status.php]:
```
Warning
In PHP 5, list() assigns the values starting with the right-most parameter. In PHP 7, list() starts with the left-most parameter.
```

Comment (Korbeil) [reply src/Client/Endpoint/Status.php]: capture d ecran 2018-05-03 a 14 44 13

It's only on push, not on simple assignation.

**Pull request merged**